### PR TITLE
fix: exit

### PR DESCRIPTION
**Pull request by colinlyguo**
### Purpose or design rationale of this PR

Better return directly when exiting, for completeness.

### PR title

Your PR title must follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) (as we are doing squash merge for each PR), so it must start with one of the following [types](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type):

- [x] fix: A bug fix

### Deployment tag versioning

Has `tag` in `common/version.go` been updated or have you added `bump-version` label to this PR?

- [x] Yes

### Breaking change label

Does this PR have the `breaking-change` label?

- [x] No, this PR is not a breaking change

### CODE REVIEW

**Comment by colinlyguo** [comment]
successfully exit:
```
INFO [07-12|21:57:48.665] L1 messenger start balance               start=10000000000000000000
INFO [07-12|21:57:48.811] watching block number                    layer=Layer2 start=451                  end=500
INFO [07-12|21:57:48.811] watching block number                    layer=Layer2 start=401                  end=450
INFO [07-12|21:57:49.477] watching block number                    layer=Layer1 start=18,308,451           end=18,308,500
INFO [07-12|21:57:49.477] watching block number                    layer=Layer1 start=18,308,401           end=18,308,450
INFO [07-12|21:57:49.981] CheckETHBalance started                  layer type=Layer2
INFO [07-12|21:57:49.982] CheckETHBalance started                  layer type=Layer1
INFO [07-12|21:57:51.130] watching block number                    layer=Layer1 start=18,308,551           end=18,308,600
INFO [07-12|21:57:51.130] watching block number                    layer=Layer1 start=18,308,501           end=18,308,550
INFO [07-12|21:57:51.132] L1 messenger start balance               start=10000000000000000000
INFO [07-12|21:57:51.901] chain-monitor server exiting success 
INFO [07-12|21:57:51.981] CheckETHBalance started                  layer type=Layer2
INFO [07-12|21:57:51.981] CheckETHBalance started                  layer type=Layer1
INFO [07-12|21:57:52.212] watching block number                    layer=Layer1 start=18,308,651           end=18,308,700
INFO [07-12|21:57:52.212] watching block number                    layer=Layer1 start=18,308,601           end=18,308,650
INFO [07-12|21:57:52.432] L1 messenger start balance               start=10000000000000000000
INFO [07-12|21:57:52.988] ContractController l1 watch the run loop exit 
INFO [07-12|21:57:53.981] CheckETHBalance started                  layer type=Layer1
INFO [07-12|21:57:53.981] CheckETHBalance started                  layer type=Layer2
INFO [07-12|21:57:54.329] L1 messenger start balance               start=10000000000000000000
INFO [07-12|21:57:55.983] CheckETHBalance started                  layer type=Layer2
INFO [07-12|21:57:55.984] CheckETHBalance started                  layer type=Layer1
INFO [07-12|21:57:56.322] L1 messenger start balance               start=10000000000000000000
INFO [07-12|21:57:57.983] CheckETHBalance started                  layer type=Layer1
INFO [07-12|21:57:57.983] CheckETHBalance started                  layer type=Layer2
INFO [07-12|21:57:58.344] L1 messenger start balance               start=10000000000000000000
INFO [07-12|21:57:58.924] checking l2 withdraw roots               start=401                  end=500
INFO [07-12|21:57:58.947] ContractController l2 watch the run loop exit 
INFO [07-12|21:57:58.947] CrossChainController l2 the run loop exit layer=Layer2
INFO [07-12|21:57:58.948] CrossChainController l1 the run loop exit layer=Layer1
INFO [07-12|21:57:58.947] alert slack the run loop exit
```

**Review by georgehao — APPROVED** [review]

File: internal/controller/contract.go [hunk 1]
```diff
@@ -159,6 +159,7 @@ func (c *ContractController) watcherStart(ctx context.Context, client *ethclient
 			return
 		case <-c.stopL2ContractChan:
 			log.Info("ContractController l2 watch the run loop exit")
+			return
```
Comment (Thegaram) [inline internal/controller/contract.go:162]: Ok, but why do we need 2 stop channels in the first place?

Comment (colinlyguo) [reply internal/controller/contract.go:162]: you are right. I'll merge them into one.

Comment (colinlyguo) [reply internal/controller/contract.go:156]: use one channel to exit in [6bfc3b8](https://github.com/scroll-tech/chain-monitor/pull/122/commits/6bfc3b8c5e39c7463fabc3c66a1f27ba83314778).

**Review by Thegaram — DISMISSED** [review]

**Review by georgehao — APPROVED** [review]

**Pull request merged**